src/runtime/pkg/containerd-shim-v2/utils.go

```diff
@@ -22,9 +22,15 @@ import (
 )
 
 func cReap(s *service, status int, id, execid string, exitat time.Time) {
+	var pid uint32
+	if execid == "" {
+		pid = s.hpid
+	} else {
+		pid = 0
```

Comment:
Can moby handle the pid of 0 correctly?

Reply:
Yes. [This is the only place in the Moby codebase which does anything with PIDs aside from reporting their values in Engine API responses.](https://github.com/moby/moby/blob/b54af02b51b0dfe2e2863e8f1647aaa27936c274/daemon/monitor.go#L158-L161) It's also the root cause of the bug which this PR is working around.

Reply:
@corhere thank you for the clarification.